### DISCUSSION TITLE
why doesn't a pattern with `/` at the beginning and end work as I would expect?

### DISCUSSION THREAD

**Discussion by zusage** [BurntSushi/ripgrep#2463] [General] [answered]
ripgrep: Non-capturing group doesn't return a value.
regexp is:
`/foo\((\w+(?:,\s*\w+)*)\);/`
string:
`foo(var, var2, var3);`

Example of regexp (PCRE) with a successful result:
[link](https://regex101.com/r/nnc3gu/1)

**Comment by BurntSushi** [comment]
Please provide an rg command with input. Show actual output and expected output

**Comment by zusage** [comment]
```
$ echo "foo(var, var2, var3);" | rg '/foo\((\w+(?:,\s*\w+)*)\);/'
$ echo $?
1
```

**Comment by BurntSushi** [comment, marked as answer]
This is why providing the _actual_ command you're running is so critical. In this case, it has nothing to do with your regex. Instead, here you are putting slashes at the beginning and end of your regex pattern. I don't know why. Just remove them and it works:

```
$ echo "foo(var, var2, var3);" | rg 'foo\((\w+(?:,\s*\w+)*)\);'
foo(var, var2, var3);
```

If there is some documentation that led you to putting slashes in your regex pattern, I'd love to fix it. Could you please point me to where you thought it was necessary to use slashes?

**Reply by zusage** [reply]
It was my typo. My mistake. **Please excuse me** for the inconvenience caused to you.

**Comment by drazde** [comment]
Hi, I have a similar problem. I would like search a url that start with '/' (not a pattern) in the files, but I cant't find how escape character

Eg:
```
$ rg '/BurntSushi/ripgrep/discussions' 
```

**Reply by BurntSushi** [reply]
Why are you trying to escape a `/`?

It isn't a special character. There is nothing to do. It should work as is. If it isn't working as you expect, then you need to _show_ your problem, as I asked for above. And open a new question please.

**Comment by drazde** [comment]
With and without `/`
```
marco@drazde MINGW64 /c/tmp
$ rg '/ws/test' test.txt

marco@drazde MINGW64 /c/tmp
$ rg 'ws/test' test.txt
2:/ws/test
```

[test.txt](https://github.com/BurntSushi/ripgrep/files/13040104/test.txt)

**Reply by ltrzesniewski** [reply]
Looks like it's specific to Mingw, it works as expected for me in PowerShell/Windows and Ubuntu/WSL:

![image](https://github.com/BurntSushi/ripgrep/assets/7913492/93fed028-daee-4eb4-9da9-c2d11a0e6b83)

![image](https://github.com/BurntSushi/ripgrep/assets/7913492/9758c4c7-5fb3-44bf-aca5-dd17eccfd815)

**Reply by BurntSushi** [reply]
Yes, that's a cygwin problem. It's mangling your pattern because it looks like a file path.

I don't know how to work around it. Sorry.

**Comment by drazde** [comment]
> Looks like it's specific to Mingw, it works as expected for me in PowerShell/Windows and Ubuntu/WSL:

Also in Windows Terminal
```
C:\Users\Marco Andreolli\Downloads>rg "/my/ws"

C:\Users\marco\Downloads>cat test.txt
foo bar
/ws/test
C:\Users\marco\Downloads>
```

But with `cmd` it works..

**Reply by BurntSushi** [reply]
Double quotes don't work on Windows I believe. Try single quotes. Or in this case, you shouldn't need quotes at all.

**Reply by drazde** [reply]
I try both single and double,  in Windows Terminal and MINGW64 (bash for Git in Windows)
but doesn't work

**Reply by BurntSushi** [reply]
Try no quotes?

I don't know, sorry. I'm not a Windows user and I don't see anything to suggest that this is a ripgrep problem. You might have better luck in a Windows help forum or something.

**Reply by BurntSushi** [reply]
And as I mentioned, IIRC, MINGW64 mangles things that look like paths. There's really nothing that I'm aware of that ripgrep can feasibly do about that. I also don't know how to work-around it.

**Reply by drazde** [reply]
Also without quote doesn't work; it's strange but i can do a workaround. 
I thought I had done something wrong
